src/languageserverinstance.jl

```diff
@@ -259,7 +259,7 @@ function Base.run(server::LanguageServerInstance)
     trigger_symbolstore_reload(server)
 
     @async try
-        while true
+        while isopen(server.jr_endpoint)
             msg = JSONRPC.get_next_message(server.jr_endpoint)
             put!(server.combined_msg_queue, (type = :clientmsg, msg = msg))
         end
```

Comment:
This strikes me as a race condition. Don't we have to make sure that `get_next_message` also doesn't get stuck if the endpoint is closed? And if we do that, we might as well not have the `isopen` call in the `while` loop because that situation would already be handled by the `get_next_message`?